x-pack/plugins/endpoint/public/applications/endpoint/lib/saga.ts

```diff
@@ -125,5 +147,19 @@ export function createSagaMiddleware(saga: Saga): Middleware {
     };
   }
 
+  middleware.start = () => {
+    runSaga();
+  };
+
+  middleware.stop = () => {
+    running = false;
+
+    // In order to actually stop the iterator instances, we need to force the loop to run one last time
```

Comment:
After setting `running` to `false`, the following actions won't have any effect, right? So why do we need to enqueue another action?

Reply:
hi @peluja1012 - correct, this action will have no effect other than to force the loop to exit. The generator loop is paused `await`ing the resolution of the current pending Promise. So when this action here is queued, that promise will resolve and the loop will exit out..

Reply:
@peluja1012 - Rob had suggested perhaps using a second Promise to control the "stop action" and use `Promise.race` inside of the `while (running)` loop. Its a another approach and if that seems clearer to all, I can implement it that way.